plugins/modules/vault_token_create.py

```diff
@@ -192,17 +189,17 @@ def run_module():
         module.exit_json(changed=changed, login={'auth': {'client_token': None}})
 
     if module.adapter.get_option('orphan'):
-        # this method is deprecated, but it's the only way through hvac to get
-        # at the /create-orphan endpoint at this time.
-        # See: https://github.com/hvac/hvac/issues/758
         try:
-            response = client.create_token(orphan=True, **legacy_options)
+            # this method was added in hvac 1.0.0
+            # See: https://github.com/hvac/hvac/pull/869
+            response = client.auth.token.create_orphan(**orphan_options)
         except AttributeError:
-            module.warn("'create_token' method was not found. Attempting method that requires root privileges.")
+            # this method was removed in hvac 1.0.0
+            # See: https://github.com/hvac/hvac/issues/758
+            response = client.create_token(orphan=True, **orphan_options)
```

Comment:
Shouldn't the above `try`/`except` be wrapped in another `try:`, so that the below `except:` is actually called if the above `client.create_token(...)` call raises an exception?

Reply:
Yes, I believe you are correct, good catch, thank you!

Reply:
https://github.com/ansible-collections/community.hashi_vault/pull/303/commits/64861315fe8a45249f441101f22ea6b0b896a8d3 adds the tests to catch this oversight (and this should be reflected in failed unit tests in CI). Next commit will have the fixed code making those tests pass.